supervision/dataset/formats/coco.py

```diff
@@ -150,7 +152,7 @@ def load_coco_annotations(
     images = {}
     annotations = {}
 
-    for coco_image in coco_images:
+    for coco_image in get_dataset_loading_tqdm_iterable(coco_images):
```

Comment:
We could directly use tqdm without separate function like this: 

`for coco_image in tqdm(coco_images, desc="Loading Dataset: "):
`

Reply:
This is done keeping extensibility in mind. The function is defined in separate file so that it's reusable for other dataset loading :)

Reply:
hi @mayankagarwals 👋🏻 I'm with @artyaltanzaya. It is awesome that you think about reusability, but like everything in programming, it is all about balance. We want to extract a single line of code into a separate function.

Reply:
> it is all about balance

That I agree with. My rationale is we will be duplicating this 

```
tqdm(coco_images, desc="Loading Dataset: ")
```
across coco, pascal voc and any other dataset we support/plan to support. 
Even if we assume tqdm is stable and won't make any breaking changes, for changing the description or adding any other argument, it needs to be repeated per dataset

To add to this, we will have tqdm imports in every file we plan on introducing progress tasks to  : Image/Video saving/processing ops, Dataset saving/processing ops, etc.

Reply:
Current idea is to expose functions like 

```
def get_dataset_loading_tqdm_iterable
def get_dataset_saving_tqdm_iterable
```

You will see that get_dataset_loading_tqdm_iterable is exposed in a util file in this pull request

Which will be reused for these specific use cases. The idea is that we will need similar tqdm instances for specific tasks like dataset loading. 

Again, this is a new repository for me, so I can't strongly comment on what's the best way. Both will work. More than happy to remove the function for now :)

Reply:
For example: 

Once we are done this PR, we will want to introduce the same for pascal voc. The change will be 

```python 
    for voc_image in get_dataset_loading_tqdm_iterable(voc_images):
```

instead of 

```python
    for voc_image in tqdm(voc_images, desc="Loading Dataset: "): 

```

Reply:
Anyway, we are pausing work on this (refer : https://github.com/roboflow/supervision/issues/183)